packages/miniflare/src/plugins/core/node-compat.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * We can provide Node.js compatibility in a number of different modes:
+ * - "legacy" - this mode adds compile-time polyfills that are not well maintained and cannot work with workerd runtime builtins.
+ * - "als": this mode tells the workerd runtime to enable only the Async Local Storage builtin library (accessible via `node:async_hooks`).
+ * - "v1" - this mode tells the workerd runtime to enable some Node.js builtin libraries (accessible only via `node:...` imports) but no globals.
+ * - "v2" - this mode tells the workerd runtime to enable more Node.js builtin libraries (accessible both with and without the `node:` prefix)
+ *   and also some Node.js globals such as `Buffer`; it also turns on additional compile-time polyfills for those that are not provided by the runtime.
+ *  - null - no Node.js compatibility.
+ */
+export type NodeJSCompatMode = "legacy" | "als" | "v1" | "v2" | null;
```

Comment:
Just a thought, not asking any change here.

I think we add to confusion with those names.

It is already confusing to have `nodejs_compat_v2` equivalent to `nodejs_compat` && `compatibility_date >= "2024-09-23"`

But naming this mode "v2" is adding unnecessary confusion. I think a descriptive name could have been better than "legacy", "v1, "v2" (and maybe "none" instead of "null").

No blaming anyone here - there are probably historic reasons. But something to keep in mind for the future.

Reply:
Actually I disagree here. I tried out various different descriptive names but in order to describe the two versions adequately they became way to verbose and in any case didn't really make much difference.

We have always talked about the two versions of runtime nodejs_compat
- v1 only handles importing built-ins prefixed with `node:` and doesn't add the globals
- v2 handles non-prefixed imports, adds globals, and further Wrangler bundles in additional polyfills.

Moreover the temporary flag to turn v2 on was indeed called `nodejs_compat_v2` not a more descriptive name.